### ISSUE TITLE
[tg-nonlin] EXC_BAD_ACCESS in api_context.cpp

### ISSUE THREAD

**Issue by izlatkin** [izlatkin/solidity_testgen#1]
TG-nonlin can not parse all solidity encoded files (with and without ADT) 
```
/Users/ilyazlatkin/CLionProjects/aeval/cmake-build-debug/tools/nonlin/tgnonlin --inv-mode 0 --no-term --keys 4271,13242 /Users/ilyazlatkin/CLionProjects/blockchain_exp/control_flow/branches_with_return/triple_nested_if/triple_nested_if_wo_adt.smt2
(error "line 3 column 28: error setting ':produce-proofs', option value cannot be modified after initialization")
Exception: EXC_BAD_ACCESS (code=1, address=0x8)
```
<img width="801" alt="image" src="https://user-images.githubusercontent.com/16242682/173406929-a51fa58c-729f-46a3-9e91-44c284c3848b.png">

**Comment by izlatkin** [comment]
[triple_nested_if_wo_adt.smt2.txt](https://github.com/izlatkin/solidity_testgen/files/8893124/triple_nested_if_wo_adt.smt2.txt)
[triple_nested_if.smt2.txt](https://github.com/izlatkin/solidity_testgen/files/8893125/triple_nested_if.smt2.txt)

**Comment by izlatkin** [comment]
fixed 9fbcd8d8d28f185f7607ca96f5eccc78264bc06f
